### PR TITLE
Upgrade from `@web3-js/scrypt-shim` to `scrypt-js`

### PR DESCRIPTION
**Pull request by ryanio**
## Description

This PR upgrades `scrypt` from [`@web3-js/scrypt-shim`](https://github.com/web3-js/scrypt-shim) to [`scrypt-js`](https://github.com/ricmoo/scrypt-js), a pure JavaScript implementation.

## Type of change

- [x] Dependency upgrade

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run dtslint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:unit` with success.
- [ ] I ran `npm run test:cov` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build-all` and tested the resulting files from `dist` folder in a browser.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have tested my code on the live network.

### CODE REVIEW

**Review by holgerd77 — APPROVED** [review]
Looks good from my side, eventually wait for an additional comment from @cgewecke before merge.

**Review by cgewecke — APPROVED** [review]
Looks perfect!

**Pull request merged**
